### PR TITLE
[TT-10016] Added support of analytics plugin

### PR DESCRIPTION
**Pull request by komalsukhani**
## Description

Added support of analytics plugin

## Related Issue

[TT-10016](https://tyktech.atlassian.net/browse/TT-10016)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [x] Make sure you are updating [CHANGELOG.md](https://github.com/TykTechnologies/tyk-operator/blob/master/CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [x] If you've changed API models, please update CRDs.
  - [x] `make manifests`
  - [x] `make helm`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `gofmt -s -w .`
  - [x] `go vet ./...`
  - [x] `golangci-lint run`

[TT-10016]: https://tyktech.atlassian.net/browse/TT-10016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/model/api.go [hunk 1]
```diff
@@ -841,6 +841,17 @@ type APIDefinitionSpec struct {
 	EnableDetailedRecording *bool `json:"enable_detailed_recording,omitempty"`
 
 	GraphQL *GraphQLConfig `json:"graphql,omitempty"`
+
+	// AnalyticsPlugin is used to configure analytics plugin which enables editing or removal of all parts of analytics
+	// records, raw request and responses recorded by Tyk at the gateway level
+	// +optional
+	AnalyticsPlugin AnalyticsPluginConfig `bson:"analytics_plugin" json:"analytics_plugin,omitempty"`
```
Comment (buraksekili) [inline api/model/api.go:848]: I think we do not need `bson` tags
```suggestion
	AnalyticsPlugin AnalyticsPluginConfig `json:"analytics_plugin,omitempty"`
```

File: api/model/api.go [hunk 2]
```diff
@@ -841,6 +841,17 @@ type APIDefinitionSpec struct {
...
+}
+
+type AnalyticsPluginConfig struct {
+	Enabled    bool   `bson:"enable" json:"enable,omitempty"`
+	PluginPath string `bson:"plugin_path" json:"plugin_path,omitempty"`
+	FuncName   string `bson:"func_name" json:"func_name,omitempty"`
```
Comment (buraksekili) [inline api/model/api.go:854]:
```suggestion
	Enabled    bool   `json:"enable,omitempty"`
	PluginPath string `json:"plugin_path,omitempty"`
	FuncName   string `json:"func_name,omitempty"`
```

File: api/model/api.go [hunk 3]
```diff
@@ -841,6 +841,17 @@ type APIDefinitionSpec struct {
...
 	GraphQL *GraphQLConfig `json:"graphql,omitempty"`
+
+	// AnalyticsPlugin is used to configure analytics plugin which enables editing or removal of all parts of analytics
+	// records, raw request and responses recorded by Tyk at the gateway level
+	// +optional
+	AnalyticsPlugin AnalyticsPluginConfig `bson:"analytics_plugin" json:"analytics_plugin,omitempty"`
+}
+
+type AnalyticsPluginConfig struct {
+	Enabled    bool   `bson:"enable" json:"enable,omitempty"`
```
Comment (buraksekili) [inline api/model/api.go:852]: Is `enabled` field optional?

Comment (komalsukhani) [reply api/model/api.go]: Nope. Will fix it

**Review by buraksekili — COMMENTED** [review]
can you please check why the integration tests failed? could it be related that the analytics plugin was introduced in v4.1.0?

**Comment by komalsukhani** [comment]
@buraksekili Resolved everything! Can you have final look?

**Pull request merged**